spec/std/yaml/yaml_pull_parser_spec.cr

```diff
@@ -95,5 +106,11 @@ module YAML
         end
       end
     end
+
+    assert_raw %(hello)
+    assert_raw %("hello"), %(hello)
+    assert_raw %(["hello"])
+    assert_raw %(["hello","world"])
+    assert_raw %({"hello":"world"})
```

Comment:
`read_raw` actually produces a JSON string that, because YAML is a superset of JSON, will parse fine. It doesn't return the original string as I don't know how to do it. I'll try the case @spalladino had, see if it can be parsed with a Union, and then I might add a few more specs before merging.